core/contract.py

```
@@ -0,0 +1,133 @@
+                logger.error(f"Contract Execution Failed: {result.get('error')}")
+                return False
+
+            # Commit updated storage only after successful execution
+            self.state.update_contract_storage(
+                contract_address,
+                result["storage"]
+            )
+
+            return True
+
+        except Exception as e:
```

Ruff (0.15.0)

[error] Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)